packages/starlight/integrations/expressive-code/index.ts

```diff
@@ -60,56 +61,55 @@ export type StarlightExpressiveCodeOptions = Omit<AstroExpressiveCodeOptions, 't
 	useStarlightUiThemeColors?: boolean | undefined;
 };
 
-export const starlightExpressiveCode = ({
-	astroConfig,
+type StarlightEcIntegrationOptions = {
+	starlightConfig: StarlightConfig;
+	useTranslations?: ReturnType<typeof createTranslationSystemFromFs> | undefined;
+};
+
+export function getStarlightEcConfigPreprocessor({
```

Comment:
Given we technically export this publicly (even if only for internal use really), it might be helpful to have a small JSDoc comment explaining that (feel free to tweak this suggestion to something you feel represents this better):

```suggestion
/**
 * Create an Expressive Code configuration preprocessor based on Starlight config.
 * Used internally to set up Expressive Code and by the `<Code>` component.
 */
export function getStarlightEcConfigPreprocessor({
```